android/src/main/java/com/amplitude/android/internal/fragments/AutocaptureFragmentLifecycleCallbacks.kt

```diff
@@ -0,0 +1,45 @@
+package com.amplitude.android.internal.fragments
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import com.amplitude.android.utilities.DefaultEventUtils.Companion.screenName
+import com.amplitude.android.utilities.DefaultEventUtils.EventProperties
+import com.amplitude.android.utilities.DefaultEventUtils.EventTypes
+import com.amplitude.common.Logger
+
+@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+internal class AutocaptureFragmentLifecycleCallbacks(
+    private val track: TrackEventCallback,
+    private val logger: Logger
+) : FragmentManager.FragmentLifecycleCallbacks() {
+    override fun onFragmentResumed(fm: FragmentManager, f: Fragment) {
+        super.onFragmentResumed(fm, f)
+
+        val className = f.javaClass.canonicalName ?: f.javaClass.simpleName ?: null
+        val fragmentIdentifier = runCatching {
+            // The id could be the `android:id` value supplied in a layout or the container view ID
+            // supplied when adding the fragment.
+            f.resources.getResourceEntryName(f.id)
+        }.onFailure {
+            logger.error("Failed to get resource entry name: $it")
+        }.getOrNull()
+        val screenName = runCatching {
+            f.activity?.screenName
+        }.onFailure {
+            logger.error("Failed to get screen name: $it")
+        }.getOrNull()
+        val fragmentTag = f.tag
+
+        track(
+            EventTypes.FRAGMENT_VIEWED,
+            mapOf(
+                EventProperties.FRAGMENT_CLASS to className,
+                EventProperties.FRAGMENT_IDENTIFIER to fragmentIdentifier,
+                EventProperties.SCREEN_NAME to screenName,
+                EventProperties.FRAGMENT_TAG to fragmentTag,
+            )
+        )
```

Comment:
Could you update the PR description to add
- which event properties are tracked with in the new event `[Amplitude] Fragment Viewed`
- if event properties can be null and when